### PR TITLE
Balder/remove steadytimestamp 2

### PR DESCRIPTION
**Pull request by baldersheim**
@vekterli PR

### CODE REVIEW

**Review by vekterli — APPROVED** [review]
:+1: some comments, mostly regarding floating point vs integer arithmetic on durations.

File: searchcore/src/tests/grouping/grouping.cpp [hunk 1]
```diff
@@ -472,24 +473,24 @@ TEST_F("test session timeout", DoomFixture()) {
     SessionId id1("foo");
     SessionId id2("bar");
 
-    GroupingContext initContext1(f1.clock, SteadyTimeStamp(10));
-    GroupingContext initContext2(f1.clock, SteadyTimeStamp(20));
+    GroupingContext initContext1(f1.clock, steady_time(duration(10)));
+    GroupingContext initContext2(f1.clock, steady_time(duration(20)));
```
[the remark this answers is not in the document]
Comment (baldersheim) [reply searchcore/src/tests/grouping/grouping.cpp]: Fixed

File: searchcore/src/vespa/searchcore/proton/server/storeonlydocsubdb.cpp [hunk 2]
```diff
@@ -125,7 +125,7 @@ StoreOnlyDocSubDB::StoreOnlyDocSubDB(const Config &cfg, const Context &ctx)
       _subDbType(cfg._subDbType),
       _fileHeaderContext(*this, ctx._fileHeaderContext, _docTypeName, _baseDir),
       _lidReuseDelayer(),
-      _commitTimeTracker(TimeStamp::Seconds(3600.0)),
+      _commitTimeTracker(vespalib::from_s(3600)),
```
[the remark this answers is not in the document]
Comment (baldersheim) [reply searchcore/src/vespa/searchcore/proton/server/storeonlydocsubdb.cpp]: Fixed

File: staging_vespalib/src/vespa/vespalib/util/doom.h [hunk 3]
```diff
@@ -8,22 +8,22 @@ namespace vespalib {
 
 class Doom {
 public:
-    Doom(const vespalib::Clock &clock, fastos::SteadyTimeStamp doom)
+    Doom(const Clock &clock, steady_time doom)
         : Doom(clock, doom, doom, false)
     {}
-    Doom(const vespalib::Clock &clock, fastos::SteadyTimeStamp softDoom,
-         fastos::SteadyTimeStamp hardDoom, bool explicitSoftDoom);
+    Doom(const Clock &clock, vespalib::steady_time softDoom,
```
[the remark this answers is not in the document]
Comment (baldersheim) [reply staging_vespalib/src/vespa/vespalib/util/doom.h]: Fixed

File: staging_vespalib/src/tests/rusage/rusage_test.cpp [hunk 4]
```diff
@@ -30,12 +30,12 @@ Test::testRUsage()
...
     EXPECT_EQUAL(diff.toString(), r2.toString());
     {
-        RUsage then = RUsage::createSelf(fastos::SteadyTimeStamp(7));
+        RUsage then = RUsage::createSelf(steady_time(duration(7)));
         RUsage now = RUsage::createSelf();
         EXPECT_NOT_EQUAL(now.toString(), then.toString());
     }
     {
-        RUsage then = RUsage::createChildren(fastos::SteadyTimeStamp(1337583));
+        RUsage then = RUsage::createChildren(steady_time(duration(1337583)));
```
[the remark this answers is not in the document]
Comment (baldersheim) [reply staging_vespalib/src/tests/rusage/rusage_test.cpp]: Fixed

File: staging_vespalib/src/tests/clock/clock_benchmark.cpp [hunk 5]
```diff
@@ -120,7 +123,7 @@ void benchmark(const char * desc, FastOS_ThreadPool & pool, uint64_t samples, ui
             count[i] += sampler->_count[i];
         }
     }
-    printf("%s: Took %ld clock samples in %2.3f with [%ld, %ld, %ld] counts\n", desc, samples, (fastos::ClockSteady::now() - start).sec(), count[0], count[1], count[2]);
+    printf("%s: Took %ld clock samples in %2.3f with [%ld, %ld, %ld] counts\n", desc, samples, to_s(steady_clock ::now() - start), count[0], count[1], count[2]);
```
[the remark this answers is not in the document]
Comment (baldersheim) [reply staging_vespalib/src/tests/clock/clock_benchmark.cpp]: Fixed

File: searchlib/src/tests/features/featurebenchmark.cpp [hunk 6]
```diff
@@ -648,8 +648,8 @@ Benchmark::Main()
         std::cout << "feature case '" << cfg.getCase() << "' is not known" << std::endl;
     }
 
-    std::cout << "TET:  " << _sample.ms() << " (ms)" << std::endl;
-    std::cout << "ETPD: " << std::fixed << std::setprecision(10) << _sample.ms() / cfg.getNumRuns() << " (ms)" << std::endl;
+    std::cout << "TET:  " << vespalib::count_ms(_sample) << " (ms)" << std::endl;
+    std::cout << "ETPD: " << std::fixed << std::setprecision(10) << vespalib::count_ms(_sample) / cfg.getNumRuns() << " (ms)" << std::endl;
```
[the remark this answers is not in the document]
Comment (baldersheim) [reply searchlib/src/tests/features/featurebenchmark.cpp]: Fixed.

File: searchlib/src/tests/attribute/benchmark/attributeupdater.h [hunk 7]
```diff
@@ -70,16 +70,16 @@ class AttributeUpdaterStatus
...
-        return _totalUpdateTime / _numDocumentUpdates;
+        return vespalib::count_ms(_totalUpdateTime) / _numDocumentUpdates;
     }
     double valueUpdateThroughput() const {
-        return _numValueUpdates * 1000 / _totalUpdateTime;
+        return _numValueUpdates * 1000 / vespalib::count_ms(_totalUpdateTime);
     }
     double avgValueUpdateTime() const {
-        return _totalUpdateTime / _numValueUpdates;
+        return vespalib::count_ms(_totalUpdateTime) / _numValueUpdates;
```
[the remark this answers is not in the document]
Comment (baldersheim) [reply searchlib/src/tests/attribute/benchmark/attributeupdater.h]: Fixed

File: searchlib/src/tests/attribute/benchmark/attributesearcher.h [hunk 8]
```diff
@@ -39,17 +39,17 @@ class AttributeSearcherStatus
...
-        std::cout << "<total-search-time>" << _totalSearchTime << "</total-search-time>" << std::endl; // ms
+        std::cout << "<total-search-time>" << vespalib::count_ms(_totalSearchTime) << "</total-search-time>" << std::endl; // ms
         std::cout << "<avg-search-time>" << avgSearchTime() << "</avg-search-time>" << std::endl; // ms
         std::cout << "<search-throughput>" << searchThroughout() << "</search-throughput>" << std::endl; // per/sec
         std::cout << "<total-hit-count>" << _totalHitCount << "</total-hit-count>" << std::endl;
         std::cout << "<avg-hit-count>" << avgHitCount() << "</avg-hit-count>" << std::endl;
     }
     double avgSearchTime() const {
-        return _totalSearchTime / _numQueries;
+        return double(vespalib::count_ms(_totalSearchTime)) / _numQueries;
```
[the remark this answers is not in the document]
Comment (baldersheim) [reply searchlib/src/tests/attribute/benchmark/attributesearcher.h]: Fixed

File: searchcorespi/src/vespa/searchcorespi/index/warmupindexcollection.cpp [hunk 9]
```diff
@@ -81,7 +80,7 @@ WarmupIndexCollection::toString() const
 
 WarmupIndexCollection::~WarmupIndexCollection()
 {
-    if (_warmupEndTime != fastos::SteadyTimeStamp::ZERO) {
+    if (_warmupEndTime != vespalib::steady_time()) {
```
[the remark this answers is not in the document]
Comment (baldersheim) [reply searchcorespi/src/vespa/searchcorespi/index/warmupindexcollection.cpp]: I suggest we discuss that with @havardpe separately.

**Pull request merged**